### PR TITLE
Update quickstarts for 8.16

### LINKED ISSUE #4534: [Request]: Include on-premises Elasticsearch cluster as an option in all quickstart guides
### Description

When applicable we should also include on-prem clusters in the `prerequisites` section of the quickstarts.

This only applies to Stateful docs.

### Resources

N/A

### Which documentation set does this change impact?

Stateful only

### Feature differences

N/A

### What release is this request related to?

N/A

### Collaboration model

The documentation team

### Point of contact.

**Main contact:** @eedugon 

**Stakeholders:** @bmorelli25

### PR DESCRIPTION
**Pull request by dedemorton**
## Description
Updates the Observability quickstarts for version 8.16.0:

* Applies requested changes.
* Adds a few consistency edits.
* Updates screenshots to match 8.16.

## Preview links

Please remember to also review the changed lines shown in the **Files changed** tab.

**Stateful:**

* https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/observability/master/quickstart-monitor-hosts-with-elastic-agent.html
* https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/observability/master/monitor-k8s-logs-metrics-with-elastic-agent.html
* https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/observability/master/monitor-k8s-otel-edot.html
* https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/observability/master/collect-data-with-aws-firehose.html

**Serverless:**

* https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/serverless/current/observability-quickstarts-monitor-hosts-with-elastic-agent.html
* https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/serverless/current/observability-quickstarts-k8s-logs-metrics.html
* https://observability-docs_bk_4582.docs-preview.app.elstc.co/guide/en/serverless/current/collect-data-with-aws-firehose.html

### Documentation sets edited in this PR

_Check all that apply._

- [x] Stateful (`docs/en/observability/*`)
- [x] Serverless (`docs/en/serverless/*`)
- [ ] Integrations Developer Guide (`docs/en/integrations/*`)
- [ ] None of the above

### Related issue
Closes #4534, #4562, and #4563

## Checklist

- [ ] Product/Engineering Review
- [ ] Writer Review

### Follow-up tasks

_Select one._

* This PR does _not_ need to be ported to another doc set because:
  - [ ] The concepts in this PR only apply to one doc set (serverless _or_ stateful)
  - [x] The PR contains edits to both doc sets (serverless _and_ stateful)
* This PR needs to be ported to another doc set:
  - [ ] Port to stateful docs: \<link to PR or tracking issue>
  - [ ] Port to serverless docs: \<link to PR or tracking issue>

### CODE REVIEW

**Comment by alaudazzi** [comment]
@dedemorton @eedugon 
Specifically to the section "Collect your data", I would remove the screenshot as it adds no added value to the short list of steps.
More in general, screenshots generate lots of maintenance, especially when documenting new features, as the UI is supposed to change frequently.

**Comment by eedugon** [comment]
I like the screenshots for this type of docs, at least to get an overview of what you are going to see when you follow the doc, even if it could get a bit outdated at some point.

But whatever you prefer, I completely understand the negative aspects of using screenshots, and as @alaudazzi says, the `Collect data` specifically doesn't add much value besides helping the user to determine that they are in the right place.

In anyway I would suggest to double check with PMs before removing the screenshots. If everybody agrees I'm totally ok with removing them too.

If the conversation is around the format of the screenshots of the `collect data` section then I prefer screenshots that show a good portion of the screen, like the one included in https://www.elastic.co/guide/en/observability/current/monitor-k8s-otel-edot.html#_collect_your_data_3 more than the capture of a very small part of the screen that ends up quite big and `zoomed` in the doc, like https://www.elastic.co/guide/en/observability/current/collect-data-with-aws-firehose.html#_collect_your_data_4.

In the second case I'd try at least to ensure the image doesn't ends up being too big on the final doc.

**Comment by dedemorton** [comment]
Thanks for you input. Since we're split in the middle, for the review, I think I will include the screenshot (using the version that provides context) and ask PMs to make a case for keeping it. If the onboarding experience is more stable because most of the quickstarts are GA, then maybe it won't change as much going forward.

File: docs/en/observability/images/quickstart-aws-firehose-entry-point.png
Comment (dedemorton) [inline docs/en/observability/images/quickstart-aws-firehose-entry-point.png:1]: I'm really interested in hearing if reviewers think these screen captures provide sufficient value to justify the cost of keeping them up-to-date over time, especially given that this UI has been quite volatile.

**Review by akhileshpok — DISMISSED** [review]
The doc covering the quickstart onboarding workflow for Firehose in serverless does not mention that this functionality is currently in Tech Preview.

**Review by thomheymann — APPROVED** [review]

**Comment by dedemorton** [comment]
>The doc covering the quickstart onboarding workflow for Firehose in serverless does not mention that this functionality is currently in Tech Preview. 

Good catch! The preview tag must have gotten removed when we did a general cleanup of preview tags for GA. (In some cases, the tag was doing double duty.)

File: docs/en/observability/images/quickstart-aws-firehose-entry-point.png
Comment (dedemorton) [reply docs/en/observability/images/quickstart-aws-firehose-entry-point.png:1]: I think I will leave the screenshots for now, and we can remove them later if we see more volatility.

**Review by mdbirnstiehl — APPROVED** [review]
🦭 👍

**Pull request merged**